### ISSUE TITLE
salt-cloud Windows Openstack not using Password

### ISSUE THREAD

**Issue by Linuturk** [saltstack/salt#20383]
When attempting to bootstrap a Windows server on openstack (Rackspace), the winexe commands aren't using the password supplied by the API. I remember fixing this in an older version of salt, so this appears to be a regression.

```
[INFO    ] Running command under pid 6000: 'winexe -U Administrator% //23.253.158.252 "sc query winexesvc"'
[DEBUG   ] Executing command(PID 6000): 'winexe -U Administrator% //23.253.158.252 "sc query winexesvc"'
[DEBUG   ] Return code was 1
```

Here's where I fixed this before: https://github.com/saltstack/salt/pull/14748

```
# salt --versions
           Salt: 2014.7.1
         Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
 msgpack-python: 0.3.0
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.0.1
           RAET: Not Installed
            ZMQ: 4.0.4
           Mako: 0.9.1
```

Salt-Minion-2014.7.1-AMD64-Setup.exe

**Comment by rallytime** [comment]
@Linuturk Yeah, it looks like your original fix went into the develop branch after the 2014.7 branch was cut. I think this is fixed on the 2015.2 and develop branches. Are you in a place where you can test that and let us know if the problem you're seeing exists there?

I am pretty sure that #14834 is the pull request that should be backported to 2014.7 to fix it, but I just want to be sure. Though I am also wondering if #15984 is going to be a problem here as well.

**Comment by Linuturk** [comment]
I can confirm when running salt from develop, this problem does not exist. The password is included in the winexe call:

```
[INFO    ] Running command under pid 18197: 'winexe -U Administrator%t8TKQeMZCkf6 //104.130.171.99 "sc query winexesvc"'
[DEBUG   ] Executing command(PID 18197): 'winexe -U Administrator%t8TKQeMZCkf6 //104.130.171.99 "sc query winexesvc"'
[DEBUG   ] Return code was 1
ERROR: Cannot open control pipe - NT_STATUS_INVALID_PARAMETER
```

```
# salt --versions
           Salt: 2015.2.0-361-gced4bfc
         Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
 msgpack-python: 0.3.0
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.0.1
           RAET: Not Installed
            ZMQ: 4.0.4
           Mako: Not Installed
```

**Comment by rallytime** [comment]
@Linuturk Ok great! I will backport that pull request to 2014.7 so we can at least fix that problem on the 2014.7 branch. :)

**Comment by Linuturk** [comment]
:+1:

**Comment by rallytime** [comment]
Alright, there you go! Give that a try on 2014.7 if you can when that gets merged in and let us know if that worked for you, at least for this issue. Thanks!

**Comment by rallytime** [comment]
I'm going to close this since the fix was backported to 2014.7 branch a little while ago. Please comment if this issue pops back up again for you and we can re-open.
